fhir-notification-nats/src/main/java/com/ibm/fhir/notifications/nats/impl/FHIRNotificationNATSPublisher.java

```diff
@@ -0,0 +1,157 @@
+        log.entering(this.getClass().getName(), "init");
+
+        try {
+            this.channelName = channelName;
+            String clientId = "fhir-server";
+            if (log.isLoggable(Level.FINER)) {
+                log.finer("ClusterId: " + clusterId);
+                log.finer("Channel name: " + channelName);
+                log.finer("Server: " + server);
+            }
+
+            // Make sure that the properties file contains the servers property at a minimum.
+            if (server == null || clusterId == null) {
+                throw new IllegalStateException("Config property missing from the NATS connection properties.");
+            }
+
+            // Create the NATS connection
+            Options streamingOptions = new Options.Builder().natsUrl(server).clusterId(clusterId).clientId(clientId).build();
+            StreamingConnectionFactory cf = new StreamingConnectionFactory(streamingOptions);
+            sc = cf.createConnection();
+
+            // Create the publish callback
+            acb = new AckHandler() {
+                @Override
+                public void onAck(String nuid, Exception ex) {
+                    log.finer("Received ACK for guid: " + nuid);
+                    if (ex != null) {
+                        log.log(Level.SEVERE, "Error in server ack for guid " + nuid + ": " + ex.getMessage(), ex);
+                    }
+                    // TODO: Do something else here
+                }
+            };
+
+            // Register this NATS implementation as a "subscriber" with our Notification Service.
+            // This means that our "notify" method will be called when the server publishes an event.
+            service.subscribe(this);
+            log.info("Initialized NATS publisher for channel '" + channelName + "' using server: " + server + ".");
+        } catch (Throwable t) {
+            String msg = "Caught exception while initializing NATS publisher.";
+            log.log(Level.SEVERE, msg, t);
+            throw new IllegalStateException(msg, t);
+        } finally {
+            log.exiting(this.getClass().getName(), "init");
+        }
+    }
+
+    /**
+     * Performs any necessary "shutdown" logic to disconnect from the channel.
+     */
+    public void shutdown() {
+        log.entering(this.getClass().getName(), "shutdown");
+
+        try {
+            if (log.isLoggable(Level.FINE)) {   
+                log.fine("Shutting down NATS publisher for channel: '" + channelName + "'.");
+            }
+            if (sc != null) {
+               sc.close();
+            }
+        } catch (Throwable t) {
+            String msg = "Caught exception shutting down NATS publisher for channel: '" + channelName + "'.";
+            log.log(Level.SEVERE, msg, t);
+            throw new IllegalStateException(msg, t);
+        } finally {
+            log.exiting(this.getClass().getName(), "shutdown");
+        }
+    }
+
+    /*
```

Comment:
We've been removing this javadoc as we go.  Please remove this comment
We try to stick to adding rather than autogenerating (the house is dirty, we clean as we go)